src/System.Reflection/tests/MethodInfo/MethodInfo_InvokeTests.cs

```diff
@@ -378,6 +379,44 @@ public static void TestInvokeMethod_DefaultParameter_StaticMethodWithMissingExpl
                     new object[] { 101, "value", 103.14m }));
         }
 
+        [Fact]
+        public static void TestInvokeMethod_OptionalParameter_WithExplicitValue()
+        {
+            Assert.Equal(
+                "value",
+                getMethod(typeof(DefaultParameters), "OptionalObjectParameter").Invoke(new DefaultParameters(), new object[] { "value" }));
+        }
+
+        [Fact]
+        public static void TestInvokeMethod_OptionalParameter_WithMissingValue()
+        {
+            Assert.Equal(
+                Type.Missing,
+                getMethod(typeof(DefaultParameters), "OptionalObjectParameter").Invoke(new DefaultParameters(), new object[] { Type.Missing }));
+        }
+
+        [Fact]
+        public static void TestInvokeMethod_OptionalParameterUnassingableFromMissing_WithMissingValue()
+        {
+            Assert.Throws<ArgumentException>(() => getMethod(typeof(DefaultParameters), "OptionalStringParameter").Invoke(new DefaultParameters(), new object[] { Type.Missing }));
```

Comment:
Nit: if possible, we like to pass the expected argument name as the first argument to the overload of Throws that takes one when expecting argument exceptions.  This helps to make sure the right value (which may be null) is coming through in the argument exception's property.  Not worth holding up this change, but you might want to go back and add those and do so for future cases like this.